codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/XmlMemberDeserVisitor.java

```diff
@@ -90,12 +94,13 @@ public Void booleanShape(BooleanShape shape) {
     /**
      * Consumes a single token into the variable "val", returning on any error.
      */
-    private void consumeToken(Shape shape) {
-        if (!filter.test(shape)) {
+    private void consumeToken() {
+        GoWriter writer = context.getWriter();
+        if (isXmlAttributeMember) {
```

Comment:
Can you add a comment explaining why this is treated differently

Reply:
Sure. 

The difference is that in case of Xml Attribute members, we already have the value in the root element's start element. ie. <root attr="value">

Thus, we do not need to pull out the next XML element from the body which we would do normally. ie. <root><attr>value</attr></root>